crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -174,28 +216,32 @@ impl Object {
 
     /// The Base64 encoded bcs serialization of the object's content.
     async fn bcs(&self) -> Result<Option<Base64>> {
-        if let Some(stored) = &self.stored {
-            Ok(Some(Base64::from(&stored.serialized_object)))
-        } else {
-            let bytes = bcs::to_bytes(&self.native)
-                .map_err(|e| {
-                    Error::Internal(format!(
-                        "Failed to serialize object at {}: {e}",
-                        self.address,
-                    ))
-                })
-                .extend()?;
+        let Some(native) = self.native_impl() else {
```

Comment:
This needs to have its optimised implementations returned to it:

```rust
use ObjectKind as K;
Ok(match &self.kind {
    K::WrappedOrDeleted(_) | OutsideAvailableRange => None,
    K::Live(_, stored) => Some(Base64::from(&stored.serialized_object)),
    K::Historical(_, stored) => Some(Base64::from(&stored.serialized_object)),
    K::Native(native) => {
        let bytes = bcs::to_bytes(native)
            .map_err(|e| {
                Error::Internal(format!("Failed to serialize object at {}: {e}", self.address))
            })
            .extend()?;
        Some(Base64::from(&bytes))
    }
})
```

Reply:
Wouldn't this lead to additional serde compared to passing a ref to NativeObject?

Reply:
No, it should avoid one serialization step, unless I'm missing something.

Reply:
Ah sorry, I misread where you had placed this comment. Yes indeed your suggestion will avoid a serialization step, oops and thanks!